internal/service/advancedclustertpf/model_ClusterDescription20240805.go

```diff
@@ -164,12 +168,12 @@ func convertReplicationSpecsLegacy(ctx context.Context, input *[]admin.Replicati
 			diags.AddError(errorZoneNameNotSet, errorZoneNameNotSet)
 			return &tfModels
 		}
-		numShards, ok := legacyInfo.ZoneNameNumShards[zoneName]
+		numShards, ok := apiInfo.ZoneNameNumShards[zoneName]
 		errMsg := []string{}
 		if !ok {
 			errMsg = append(errMsg, fmt.Sprintf(errorNumShardsNotSet, zoneName))
 		}
-		legacyID, ok := legacyInfo.ZoneNameReplicationSpecIDs[zoneName]
+		legacyID, ok := apiInfo.ZoneNameReplicationSpecIDs[zoneName]
```

Comment:
in auto scaling dev branch (CLOUDP-256272-auto-scaling-per-shard) we use `compressAPIObjectList` to remove duplicate zone names, can definitely see we can unify logic over to tpf implementation as well. I believe this is currently done with tfModelsSkipIndexes logic right?

Reply:
Correct